### PR TITLE
Adds documentation for sign in command.

### PR DESCRIPTION
**Pull request by samjonester**
This pull request is paired with https://github.com/rubygems/rubygems/pull/1635/

### CODE REVIEW

File: publishing.md [hunk 1]
```diff
@@ -64,16 +64,48 @@ should set this value to prevent accidental pushes to rubygems.org:
...
+After creating the account, use your email and password when signing in.
+(RubyGems saves the credentials in ~/.gem/credentials for you so you only need
+to log in once.)
+
+Alternatively, you can authenticate with a private registry by providing the `--host`
+flag while signing in.
+
+To authenticate and enable you to push version 0.1.0 of a new gem named 'squid-utils':
+
+    $ gem signin squid-utils-0.1.0.gem
```
Comment (tjouan) [inline publishing.md]: With current signin command implementation, I don't think it requires
or accept a gem as argument, so I think it should just be:

    gem signin

**Comment by colby-swandale** [comment]
ping @samjonester

**Comment by samjonester** [comment]
This PR was for a previous implementation that was never merged into Ruby
gems proper. The associated PR into Ruby gems was closed

**Comment by sonalkr132** [comment]
Sign in doc was added in #190. Thanks anyway, @samjonester

**Pull request closed without merging**